indico/modules/events/registration/client/js/components/participant-list/ParticipantTable.tsx

```diff
@@ -0,0 +1,162 @@
+    };
+
+    const direction: sortDirectionType =
+      sortColumn === column ? directions[sortDirection] : 'ascending';
+
+    const sortedData =
+      direction === null
+        ? [...table.rows]
+        : [...sortedRows].sort((a, b) => {
+            const comparedVals = [a, b].map(el =>
+              typeof column === 'string' ? el[column] : el.columns[column].text
+            );
+
+            let sortResult;
+
+            if (comparedVals[0] === comparedVals[1]) {
+              return 0;
+            } else if (comparedVals.every(el => typeof el === 'boolean')) {
+              sortResult = comparedVals[0] > comparedVals[1] ? -1 : 1;
+            } else {
+              sortResult = comparedVals[0].localeCompare(comparedVals[1]);
+            }
+
+            return sortResult * (direction === 'ascending' ? 1 : -1);
+          });
+
+    setSortColumn(column);
+    setSortDirection(direction);
+    setSortedRows(sortedData);
+  };
+
+  return visibleParticipantsCount > 0 ? (
+    <Table fixed celled sortable>
+      <TableHeader>
+        <TableRow>
+          {table.show_checkin && ( // For checkin status
+            <TableHeaderCell
+              width={2}
+              textAlign="center"
+              styleName="participant-table-row"
+              onClick={() => handleSort('checked_in')}
+              sorted={sortColumn === 'checked_in' ? sortDirection : undefined}
+            >
+              <Icon name="ticket" />
+            </TableHeaderCell>
+          )}
+          {table.headers.map((headerText: string, i: number) => {
+            const isSortable = isColumnSortable(i);
+
+            return (
+              <TableHeaderCell
+                // eslint-disable-next-line react/no-array-index-key
+                key={i}
+                width={6}
+                styleName="participant-table-row"
+                onClick={isSortable ? () => handleSort(i) : undefined}
+                sorted={isSortable && sortColumn === i ? sortDirection : undefined}
+                title={`${headerText} ${!isSortable ? '(Not sortable)' : ''}`}
+              >
+                <p>{headerText}</p>
+              </TableHeaderCell>
+            );
+          })}
+        </TableRow>
+      </TableHeader>
+      <TableBody>
+        {sortedRows.map((row: TableRowObj) => (
+          <TableRow key={row.id}>
+            {table.show_checkin && (
+              <TableCell textAlign="center">
+                {row.checked_in ? (
+                  <Icon name="check" color="green" />
+                ) : (
+                  <Icon name="close" color="red" />
+                )}
+              </TableCell>
+            )}
+            {row.columns.map((col: TableColumnObj, i: number) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <TableCell key={i} title={col.text}>
+                {col.is_picture && col.text ? (
+                  <img src={col.text} alt={col.text} styleName="cell-img" />
```

Comment:
Just commenting my own two pence: for tables which become very large in width, it might be worth considering to add a clickable full-screen preview for the image, like we do on RB for room images (i.e. you click on the image to enlarge it).

cc: @tomasr8 any thoughts?

Reply:
As we discussed on MM, this would be a nice feature but let's do it in a separate PR as this one is already pretty large

Reply:
Indeed I was thinking of adding that as well @GovernmentPlates . I'll make a list of continuation issues, including for this one

Reply:
Issue made https://github.com/indico/indico/issues/6507